custom_components/tuya_local/devices/ecostrad_iq_heating_element.yaml

```diff
@@ -0,0 +1,116 @@
+name: Ecostrad iQ Heating Element
+primary_entity:
+  entity: climate
+  icon: "mdi:radiator"
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: true
+          value: "heat"
+        - dps_val: false
+          value: "off"
+    - id: 2
+      type: string
+      name: preset_mode
+      mapping:
+        - dps_val: "auto"
+          value: Program
+          icon: "mdi:clock-outline"
+        - dps_val: "eco"
+          value: Eco
+          icon: "mdi:moon-waxing-crescent"
+        - dps_val: "hot"
+          value: Comfort
+          icon: "mdi:weather-sunny"
+        - dps_val: "only_inside"
+          value: Radiator
+          icon: "mdi:radiator"
+    - id: 16
+      type: integer
+      name: temperature
+      range:
+        min: 70
+        max: 300
+      mapping:
+        - scale: 10
+          step: 5
+    - id: 24
+      type: integer
+      name: current_temperature
+      mapping:
+        - scale: 10
+secondary_entities:
+  - entity: number
+    name: Radiator Mode Temperature

```

Comment:
If this is only valid in Radiator mode, then it may be better inside the climate entity, and linked to the temperature dp by:
```
      mapping:
         - scale: 10
           step: 5
           constraint: preset_mode
           conditions:
             - dps_val: only_inside
               value_redirect: radiator_temperature